features/steps/thens/attributes.py

```diff
@@ -117,9 +118,13 @@ def step_impl(context, inst, value_or_comparison_op:str, attribute:str=None, exp
         except Exception as e:
             raise ValueError(f"Error evaluating expression: {e}")
 
-        # Compare the attribute with the expression value, considering the precision
-        entity_contexts = geometry.recurrently_get_entity_attr(context, inst, 'IfcRepresentation', 'ContextOfItems')
-        precision = geometry.get_precision_from_contexts(entity_contexts)
+        # Compare the attribute with the expression value, considering the specified tolerance or
+        # precision of the applicable geometric context
+        if comparison_tolerance:
+            precision = float(comparison_tolerance)
```

Comment:
```suggestion
            precision = comparison_tolerance
```

^ with the above the cast is not necessary on our side